src/unit-number/unit-number-input.tsx

```diff
@@ -0,0 +1,341 @@
+import React, { useEffect, useRef, useState } from "react";
+import { StringHelper } from "../util/string-helper";
+import {
+    Container,
+    FloorInput,
+    InputContainer,
+    UnitInput,
+    UnitNumberDivider,
+} from "./unit-number-input.style";
+import { AddOnContainer } from "../input-group/input-group.style";
+import { DateInputProps } from "../date-input/types";
+
+type FieldType = "floor" | "unit" | "none";
+type ValueFieldTypes = Exclude<FieldType, "none">;
+
+export interface UnitNumberInputProps extends DateInputProps {
```

Comment:
would prefer not to extend from `DateInputProps`, especially since those properties have date input specific comments